File: attacks/sqli-error-based-lab/src/health.py
import time
from flask import jsonify
from config import LAB_NAME, DB_PATH
from db import get_conn

def health_handler():
    status_code = 200
    try:
        conn = get_conn()
        conn.execute("SELECT 1")
        conn.close()
        checks = {"database": {"status": "ok", "path": DB_PATH}}
    except Exception as e:
        status_code = 503
        checks = {"database": {"status": "error", "detail": str(e), "path": DB_PATH}}

    return jsonify({
        "status": "healthy" if status_code == 200 else "unhealthy",
        "service": LAB_NAME,
        "timestamp": time.strftime("%Y-%m-%d %H:%M:%S"),
        "checks": checks
    }), status_code

File: attacks/sqli-auth-lab/src/health.php
<?php
require_once __DIR__ . '/config.php';

header('Content-Type: application/json; charset=UTF-8');

$response = [
    'status' => 'healthy',
    'service' => $labName,
    'timestamp' => date('Y-m-d H:i:s'),
    'checks' => [
        'database' => [
            'status' => 'ok',
            'path' => $dbPath,
        ],
    ],
];

try {
    $db = getDb();
    $db->query('SELECT 1');
    if (!file_exists($dbPath)) {
        require __DIR__ . '/reset.php';
        exit;
    }
    http_response_code(200);
    echo json_encode($response, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE);
} catch (Throwable $e) {
    http_response_code(503);
    $response['status'] = 'unhealthy';
    $response['checks']['database']['status'] = 'error';
    $response['checks']['database']['message'] = $debugMode ? $e->getMessage() : 'Database unavailable';
    echo json_encode($response, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE);
}
